npm/iptm/iptm.go

```diff
@@ -253,8 +285,8 @@ func (iptMgr *IptablesManager) checkAndAddForwardChain() error {
 		return err
 	}
 	iptMgr.OperationFlag = util.IptablesInsertionFlag
-	// Reduce index for deleted AZURE-NPM chain
-	if index > 1 {
+	if !iptMgr.shouldPlaceAzureChainFirst {
+		// Reduce index for deleted AZURE-NPM chain
 		index--
 	}
 	entry.Specs = append([]string{strconv.Itoa(index)}, entry.Specs...)
```

Comment:
in entry.specs, we do not have to explicitly say index 1 while inserting, we can just give -I flag and rule will be added to top. I think that is better than having to pass in index, what if the index gets corrupted tomorrow due to a bug ? 
```go
if !iptMgr.shouldPlaceAzureChainFirst {
	// Reduce index for deleted AZURE-NPM chain
	index--
}
entry.Specs = append([]string{strconv.Itoa(index)}, entry.Specs...)
if errCode, err = iptMgr.run(entry); err != nil {
```
Change To 

```go


if !iptMgr.shouldPlaceAzureChainFirst {
	// Reduce index for deleted AZURE-NPM chain
	index--
        entry.Specs = append([]string{strconv.Itoa(index)}, entry.Specs...)
}
if errCode, err = iptMgr.run(entry); err != nil {
```

Reply:
Good point. Will change